reference/gall.md

```diff
@@ -0,0 +1,765 @@
+# Gall
+
+## Reference Agent
+
+This app stores a number for each user. You can increment your number, increment
+someone else's number, start tracking someone else's number by poking. It
+exposes a scry namespace to inspect everyone's number.
+
+### `app/example-gall.hoon`
+
+```hoon
+/-  *example-gall
+/+  default-agent, verb
+|%
++$  card  card:agent:gall
++$  note
+  $%
+    [%arvo =note-arvo]
+    [%agent [=ship name=term] =task:agent:gall]
+  ==
++$  state  [local=@ud remote=(map @p @ud)]
+--
+=|  state
+=*  sta  -
```

Comment:
Might be better practice to do versioned state here. Also makes it more obvious how upgrading can happen in the `+on-load` docs below. Then:

```hoon
+$  state-0  [%0 local=@ud remote=(map @p @ud)]
--
=|  state-0
=*  state  -
```

That way you can avoid the three-letter-word. Also, below, you're free to do just `local` instead of `local.sta`, since the `=|` here already sticks it right in your subject, and the `=*` is just useful for if you need to refer to the state in its entirety. (Though "don't call out `.state` explicitly" is more of a personal preference than a guideline. Wonder if anyone hard-disagrees though.)